src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -141,19 +139,21 @@ public static (AddRangeResult result, bool moreChildrenToRight, ProgressTracker.
 
             tree.UpdateRootHash();
 
-            if (tree.RootHash != expectedRootHash)
+            if (tree.RootHash.ValueHash256 != expectedRootHash)
             {
-                return (AddRangeResult.DifferentRootHash, true, null);
+                return (AddRangeResult.DifferentRootHash, true);
             }
 
-            var storageLock = progressTracker.GetLockObjectForPath(accountHash);
-            storageLock.ExecuteSafe(() =>
+            // This will work if all StorageRange requests share the same AccountWithPath object which seems to be the case.
+            // If this is not true, StorageRange request should be extended with a lock object.
+            // That lock object should be shared between all other StorageRange requests for same account.
+            lock(account.Account)
```

Comment:
Main change here!

Reply:
The main point of having this complexity around locking was to lock only when necessary - that is when split is done and there us a chance of actually having parallel processing of storage ranges for same account. If the overhead is negligible, I'm all for it - it's the simplest.

Reply:
On having a conversation with @Scooletz yesterday, there could be a way to optimize it easily, if there is significant overhead. Stitching only happens when there are boundary proof nodes - so can lock only if that happens.

Reply:
lock is super cheap compared to I/O so locking even when not necessary shouldn't degrade performance

Reply:
Should be fine. That said, this code path is run quite a lot, might wannna double check if the lock is significant.

Reply:
If there is no contention then lock is very fast, if there is contention, well we still have to lock. Keep in mind in the previous solution we locked on potential contention while also accessed ConcurrentDictionary one or few times.